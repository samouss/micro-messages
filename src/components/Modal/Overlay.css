.Overlay {
  visibility: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.75);
  opacity: 0;
  transition: visibility 0s 0.25s, opacity 0.25s;
}

.Overlay--visible {
  opacity: 1;
  visibility: visible;
  transition-delay: 0s;
}
