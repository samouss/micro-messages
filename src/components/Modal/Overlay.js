// @flow

import React from 'react';
import cx from 'classnames';
import './Overlay.css';

export type Props = {
  isOpen: boolean,
};

const Overlay = ({ isOpen }: Props) => (
  <div
    styleName={cx('Overlay', {
      'Overlay--visible': isOpen,
    })}
  />
);

export default Overlay;
