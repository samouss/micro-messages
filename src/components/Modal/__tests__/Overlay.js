// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Overlay from '../Overlay';

describe('<Overlay />', () => {
  const defaultProps = {
    isOpen: false,
  };

  it('expect to render the component correctly closed', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Overlay {...props}>
        <div>some content</div>
      </Overlay>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render the component correctly open', () => {
    const props = {
      ...defaultProps,
      isOpen: true,
    };

    const component = shallow(
      <Overlay {...props}>
        <div>some content</div>
      </Overlay>,
    );

    expect(component).toMatchSnapshot();
  });
});
