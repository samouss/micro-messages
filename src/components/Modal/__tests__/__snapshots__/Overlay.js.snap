// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> expect to render the component correctly closed 1`] = `
<div
  styleName="Overlay"
/>
`;

exports[`<Overlay /> expect to render the component correctly open 1`] = `
<div
  styleName="Overlay Overlay--visible"
/>
`;
